skyve-ext/src/main/java/org/skyve/impl/util/WebStatsUtil.java

```diff
@@ -2,43 +2,155 @@
-		BindUtil.set(loginRecord, "failed", Boolean.FALSE);
+		BindUtil.set(loginRecord, AppConstants.USER_NAME_ATTRIBUTE_NAME, user.getName());
+		BindUtil.set(loginRecord, LOGIN_DATE_TIME, new DateTime(System.currentTimeMillis()));
+		BindUtil.set(loginRecord, FAILED, Boolean.FALSE);
+		BindUtil.set(loginRecord, IP_ADDRESS, userIPAddress);
+
+		// Check if the IpInfo token has been set so as to get the country code
+		String ipInfoToken = UtilImpl.IP_INFO_TOKEN;
+		String countryCode = null;
+		if (ipInfoToken != null) {
+			final GeoIPService geoIPService = new GeoIPService();
+			Optional<String> countryCodeOptional = geoIPService.getCountryCodeForIP(userIPAddress);
+			if (countryCodeOptional.isPresent()) {
+				countryCode = countryCodeOptional.get();
+				BindUtil.set(loginRecord, COUNTRY, countryCode);
+			}
+		}
+		AbstractPersistence.get()
+				.save(loginRecordDocument, loginRecord);
+
+		// Get the last login record of the current user
+		DocumentQuery q = CORE.getPersistence()
+				.newDocumentQuery(loginRecordDocument);
+		q.getFilter()
+				.addEquals(AppConstants.USER_NAME_ATTRIBUTE_NAME, user.getName());
+		q.addBoundOrdering(LOGIN_DATE_TIME, SortDirection.descending);
+		AbstractPersistentBean previousLoginRecord = q.beanResult();
+
+		// Get email and name from the metadata user
+		Document userDocument = module.getDocument(customer, AppConstants.USER_DOCUMENT_NAME);
+		AbstractPersistentBean adminUserBean = CORE.getPersistence()
+				.retrieve(userDocument, user.getId());
+		String contactEmailBinding = BindUtil.createCompoundBinding(AppConstants.CONTACT_ATTRIBUTE_NAME,
+				AppConstants.EMAIL1_ATTRIBUTE_NAME);
+		String contactNameBinding = BindUtil.createCompoundBinding(AppConstants.CONTACT_ATTRIBUTE_NAME,
+				AppConstants.NAME_ATTRIBUTE_NAME);
+		String userEmail = (String) BindUtil.get(adminUserBean, contactEmailBinding);
+		String userName = (String) BindUtil.get(adminUserBean, contactNameBinding);
+
+		// Check if the ip address changed since last login and if so log the event
+		String lastIpAddress = (String) BindUtil.get(previousLoginRecord, IP_ADDRESS);
+		if (userIPAddress != lastIpAddress) {
+			SecurityUtil.log("Change of IP Address from Last Login",
+					String.format(IP_CHANGE_LOG_MESSAGE, userName, lastIpAddress, userIPAddress));
+
+		}
+
+		// Check if the country has changed since the last login and if so send the user a warning message
+		String previousCountryCode = (String) BindUtil.get(previousLoginRecord, "countryCode");
+
+		if (countryCode != null && countryCode != previousCountryCode) {
+			// Get actual country name
+			Locale locale = new Locale("", countryCode);
+			String country = locale.getDisplayCountry();
+
+			Locale previousLocale = new Locale("", previousCountryCode);
+			String previousCountry = previousLocale.getDisplayCountry();
+
+			SecurityUtil.log("User Logged in from Different Country",
+					String.format(COUNTRY_CHANGE_LOG_MESSAGE, userName, previousCountry, country));
```

Comment:
Per the Trello card, the security log should be the IP and country in a single log, so we don't have 2x the number of security logs.